endpoints/get.go

```diff
@@ -63,12 +63,55 @@ func writeGetResponse(w http.ResponseWriter, id string, value string) (error, in
 		w.Header().Set("Content-Type", "application/json")
 		w.Write([]byte(value)[len(backends.JSON_PREFIX):])
 	} else {
-		return fmt.Errorf("uuid=%s: Cache data was corrupted. Cannot determine type.", id), http.StatusInternalServerError
+		return fmt.Errorf("Cache data was corrupted. Cannot determine type."), http.StatusInternalServerError
 	}
 	return nil, http.StatusOK
 }
 
-func respondAndLogError(w http.ResponseWriter, err error, status int) {
-	log.Errorf(err.Error())
-	http.Error(w, err.Error(), status)
+// Will prefix error messages with "GET /cache" and, if uuid string list is passed, will
+// follow with the first element of it in the following fashion: "uuid=FIRST_ELEMENT_ON_UUID_PARAM".
+// Expects non-nil error
+func handleException(w http.ResponseWriter, err error, status int, uuid string) {
+	// Build message
+	msg := "GET /cache"
+	if len(uuid) > 0 {
+		msg = fmt.Sprintf("%s uuid=%s:", msg, uuid)
+	}
+	msg = fmt.Sprintf("%s %s", msg, err.Error())
+
+	// Select level
+	level := determineLogLevel(msg)
+
+	// Log and send response
+	logAtLevel(level, msg)
+	http.Error(w, msg, status)
+}
+
+func determineLogLevel(errMsg string) log.Level {
+	if strings.HasSuffix(errMsg, backends.GetKeyNotFound) {
```

Comment:
The string compare raises an eyebrow for me. We should not use string comparison for this. I think it would be better to have a custom key not found error struct and use type checking. This way the same error type can be shared among all backend stores.